osidb/models.py

```diff
@@ -723,6 +725,22 @@ def _validate_reported_date(self):
         if self.reported_dt is None:
             raise ValidationError("Flaw has an empty reported_dt")
 
+    def _validate_public_unembargo_date(self):
+        """
+        Check that an unembargo date (public date) exists if the Flaw is public
+        """
+        if not self.is_embargoed and self.unembargo_dt is None:
+            raise ValidationError("Public flaw has an empty unembargo_dt")
+
+    def _validate_future_unembargo_date(self):
```

Comment:
This check is still not correct. We want to validate that IF the flaw is embargoed AND has unembargo date THEN the date is NOT in the past. The reason is that the flaw should be made public at the point of unembargo date so when it is still embargoed the unembargo date cannot be passed.

Reply:
Sorry, I had misunderstood the story.